ddtrace/profiling/recorder.py

```diff
@@ -34,10 +34,22 @@ class Recorder(object):
 
     events = attr.ib(init=False, repr=False, eq=False)
     _events_lock = attr.ib(init=False, repr=False, factory=nogevent.DoubleLock, eq=False)
-    _pid = attr.ib(init=False, repr=False, factory=os.getpid)
 
     def __attrs_post_init__(self):
+        # type: (...) -> None
         self._reset_events()
+        forksafe.register(self._after_fork)
+
+    def _after_fork(self):
+        # type: (...) -> None
+        # NOTE: do not try to push events if the process forked
+        # This means:
+        # 1. the process has forked
+        # 2. we don't know the state of _events_lock and it might be unusable — we'd deadlock
```

Comment:
```suggestion
        # NOTE: do not try to push events if the process forked
        # This means we don't know the state of _events_lock and it might be unusable — we'd deadlock
```